indico/modules/events/registration/forms.py

```diff
@@ -407,3 +407,14 @@ def __init__(self, *args, **kwargs):
         super(ImportRegistrationsForm, self).__init__(*args, **kwargs)
         if not self.regform.moderation_enabled:
             del self.skip_moderation
+
+
+class RejectRegistrantsForm(IndicoForm):
+    rejection_reason = TextAreaField(_('Reason'), description=_("You can provide a reason for the rejection here."))
+    attach_rejection_reason = BooleanField(_('Attach reason'), [HiddenUnless('rejection_reason')],
```

Comment:
I would remove the HiddenUnless. It only triggers when the field loses focus, so its UX is not very good right now.

Just make sure that we never show an empty reason in a notification email when this is enabled but no reason is specified ;)

Reply:
I think my second point is already covered since `render_rejection_reason` doesn't render anything if there is no reason.